File: under-the-hood-of-coding-agent/log.ts
export function log(label: string, data: any) {
    console.log(`<${label}>`);
    if (typeof data === "string") {
        console.log(data);
    } else {
        console.log(JSON.stringify(data, null, 2));
    }
    console.log(`</${label}>`);
}

File: agent-playground/log.go
package main

import (
	"encoding/json"
	"fmt"
)

func logMsg(label string, data any) {
	fmt.Printf("\n<%s>\n", label)
	switch v := data.(type) {
	case string:
		fmt.Println(v)
	default:
		b, err := json.MarshalIndent(v, "", "  ")
		if err != nil {
			fmt.Println(v)
		} else {
			fmt.Println(string(b))
		}
	}
	fmt.Printf("</%s>\n", label)
}
